MemoryManagement/src/memorymanagementAlgorithm/Process.java

```diff
@@ -1,140 +1,192 @@
 package memorymanagementAlgorithm;
 
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+/**
+ *
+ * @author ahmed
+ */
+
 import java.util.*;
 import javafx.scene.paint.Color;
+//Vector<Double> list = new Vector<Double>();
 
 public class Process {
 
+    
+
+    private Color color;
     private static int Process_ID = 0;
     private int ID;
-    private Color color;
     private boolean inserted = false;
+    private int number_of_segments;
     private Vector<Segment> Segment_vector;
-
     //====================constructor============================//
-    public Process() {
+
+    public Process()// just the id
+    {
         ID = Process_ID;
         Process_ID++;
         Segment_vector = new Vector<Segment>();
+        number_of_segments = Segment_vector.size();
         color = randomColor();
     }
 
-    public Process(Segment input) {
+    public Process(Segment input)// single segment
+    {
         ID = Process_ID;
         Process_ID++;
         Segment_vector = new Vector<Segment>();
         Segment_vector.add(input);
+        number_of_segments = Segment_vector.size();
         color = randomColor();
     }
 
-    public Process(Vector<Segment> input) {
+    // use this in the gui
+    public Process(Vector<Segment> input)// vector
+    {
         ID = Process_ID;
         Process_ID++;
+        Segment_vector = new Vector<Segment>();
         Segment_vector = input;
+        number_of_segments = Segment_vector.size();
         color = randomColor();
+       
     }
 
     //===============set section =============================//
+    // to add in the gui table 
     public void add_Segment(Segment input) {
-        Segment_vector.add(input);
-        Collections.sort(Segment_vector, (a, b) -> a.getLimit() < b.getLimit() ? -1 : a.getLimit() == b.getLimit() ? 0 : 1);
+        getSegment_vector().add(input);
+        number_of_segments = getSegment_vector().size();

```

Comment:
this is a stupid redundant. every vector has its own size and its complicity is O(1). So you recount after him!

Reply:
naive